log.cpp

```diff
@@ -0,0 +1,79 @@
+#include "log.hpp"
+#include <iostream>
+#include <fstream>
+#include <ctime>
+using namespace std;
+
+Log::Log(std::string path) {
+	this->path = path;
+	logMsg.open(path);
+}
+
+ofstream& Log::getStream() {
+	return logMsg;
+}
+
+void Log::TCPconnection(remotePeerID, localPeerID) {
+	this->getStream() << date_time << ": Peer " << localPeerID << " makes a connection to Peer " << remotePeerID << ".\n";
+	logMsg.close();

```

Comment:
Once the stream is closed, you wont be able to write to it again. We need the file stream to remain open so we can write to it for the entire duration of the PeerProcess.

Create a separate function for close(), and I'll make sure the peer process calls it upon exiting